lib/saml_idp/controller.rb

```diff
@@ -44,6 +44,7 @@ def encode_authn_response(principal, opts = {})
       expiry = opts[:expiry] || 60*60
       encryption_opts = opts[:encryption] || nil
       signature_opts = opts[:signature] || {}
+      response_name_id_format = opts[:name_id_format] || saml_request.name_id_format
```

Comment:
I'm a bit confused here. `opts[:name_id_format]` is always going to be nil (opts set in https://github.com/18F/identity-idp/blob/master/app/controllers/concerns/saml_idp_auth_concern.rb#L94) and the story is to make them not have to provide the name_id_format, so it doesn't need to be in the saml_request.

How is the default (per the story, `urn:oasis:names:tc:SAML:1.1:nameid-format:persistent`) getting set?

Reply:
opts[:name_id_format] was added for the purpose of testing the saml_response with a specified name id format

This PR is a follow up PR to:
https://github.com/18F/saml_idp/pull/31

The persistent default will be added to the IDP. There is a draft PR here:
https://github.com/18F/identity-idp/pull/3912/files

Slack discussion on how this will work here:
https://gsa-tts.slack.com/archives/C0NGESUN5/p1594655231247100